File: web-socket/illchess/illchess-domain/src/main/java/pl/illchess/domain/board/exception/BoardNotFoundException.java
package pl.illchess.domain.board.exception;

import pl.illchess.domain.board.model.BoardId;
import pl.illchess.domain.commons.exception.DomainException;

import java.util.UUID;

public class BoardNotFoundException extends DomainException {
    public BoardNotFoundException(BoardId boardId) {
        super("Board with given id = %s not found".formatted(boardId.uuid()));
    }

    public BoardNotFoundException(UUID boardId) {
        super("Board with given id = %s not found".formatted(boardId));
    }
}

File: illchess-stockfish-service/illchess-stockfish/illchess-stockfish-domain/src/main/kotlin/pl/illchess/stockfish/domain/board/exception/BoardNotFoundException.kt
package pl.illchess.stockfish.domain.board.exception

import pl.illchess.stockfish.domain.board.domain.GameId
import pl.illchess.stockfish.domain.commons.NotFoundException

class BoardNotFoundException(gameId: GameId) :
    NotFoundException(String.format("Board with id  %s was not found", gameId))